database/rrddim.c

```diff
@@ -380,6 +381,9 @@ inline collected_number rrddim_set_by_pointer(RRDSET *st, RRDDIM *rd, collected_
 
     rd->collections_counter++;
 
+    collected_number v = (value >= 0) ? value : -value;
+    if(unlikely(v > rd->collected_value_max)) rd->collected_value_max = v;
```

Comment:
Do we need to update the maximum value for every data collection? I'd do it only for overflows.

Reply:
The expensive part here is the condition, not the update, since the update will only happen when the value is the max.

To check for overflow, the condition would be more complex at that point. It would require to take the `abs()` of the `last_collected_value` too and then execute a very similar condition.

Also, keeping in the structure the `collected_value_overflow_max` is somewhat niche. The simple max could have more uses in the future. This is why I chose to keep it on all metrics, of all types (not just `incremental` ones).